homeassistant/components/mysensors.py

```diff
@@ -0,0 +1,219 @@
+"""
+homeassistant.components.mysensors
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+MySensors component that connects to a MySensors gateway via pymysensors
+API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.mysensors.html
+
+
+New features:
+
+New MySensors component.
+Updated MySensors Sensor platform.
+New MySensors Switch platform.
+Multiple gateways are now supported.
+
+Configuration.yaml:
+
+mysensors:
+  port:
+    - '/dev/ttyUSB0'
+    - '/dev/ttyACM1'
+  debug: true
+  persistence: true
+  persistence_file:
+    - 'path/to/.homeassistant/mysensors.json'
+    - 'path/to/.homeassistant/mysensors2.json'
+  version: '1.5'
+
+sensor:
+  platform: mysensors
+
+switch:
+  platform: mysensors
+"""
+import logging
+
+from homeassistant.helpers import (validate_config)
+
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START,
+    EVENT_HOMEASSISTANT_STOP,
+    TEMP_CELCIUS)
+
+CONF_PORT = 'port'
+CONF_DEBUG = 'debug'
+CONF_PERSISTENCE = 'persistence'
+CONF_PERSISTENCE_FILE = 'persistence_file'
+CONF_VERSION = 'version'
+
+DOMAIN = 'mysensors'
+DEPENDENCIES = []
+REQUIREMENTS = [
+    'https://github.com/theolind/pymysensors/archive/'
+    '2aa8f32908e8c5bb3e5c77c5851db778f8635792.zip#pymysensors==0.3']
+_LOGGER = logging.getLogger(__name__)
+ATTR_PORT = 'port'
+ATTR_DEVICES = 'devices'
+ATTR_NODE_ID = 'node_id'
+ATTR_CHILD_ID = 'child_id'
+ATTR_UPDATE_TYPE = 'update_type'
+
+IS_METRIC = None
+CONST = None
+GATEWAYS = None
+EVENT_MYSENSORS_NODE_UPDATE = 'MYSENSORS_NODE_UPDATE'
+
+
+def setup(hass, config):  # noqa
+    """ Setup the MySensors component. """
+    # pylint:disable=no-name-in-module
```

Comment:
It's just in atom editor I get these lint errors. If I run script/lint, I only get these, even if I remove above lint switches:

homeassistant/components/mysensors.py:1:1: D205 1 blank line required between summary line and description (found 0)
homeassistant/components/mysensors.py:1:1: D400 First line should end with a period (not 's')
homeassistant/components/sensor/mysensors.py:1:1: D205 1 blank line required between summary line and description (found 0)
homeassistant/components/sensor/mysensors.py:1:1: D400 First line should end with a period (not 's')
homeassistant/components/sensor/mysensors.py:71:1: D203 1 blank line required before class docstring (found 0)
homeassistant/components/switch/mysensors.py:1:1: D205 1 blank line required between summary line and description (found 0)
homeassistant/components/switch/mysensors.py:1:1: D400 First line should end with a period (not 's')
homeassistant/components/switch/mysensors.py:59:1: D203 1 blank line required before class docstring (found 0)
************* Module homeassistant.components.sensor.mysensors
E: 54,21: Class 'SetReq' has no '__members__' member (no-member)
E: 54,21: Class 'SetReq' has no '__members__' member (no-member)